src/EventListener/CatchAttributesListener.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace FOS\HttpCacheBundle\EventListener;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ConfigurationInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\ControllerResolver;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\Kernel;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+if (Kernel::MAJOR_VERSION >= 5) {
+    class_alias(RequestEvent::class, 'FOS\HttpCacheBundle\EventListener\AttributeRequestEvent');
+} else {
+    class_alias(GetResponseEvent::class, 'FOS\HttpCacheBundle\EventListener\AttributeRequestEvent');
+}
+
+class CatchAttributesListener implements EventSubscriberInterface
+{
+    /**
+     * @var ControllerResolver
+     */
+    private $controllerResolver;
+
+    public function __construct(ControllerResolver $controllerResolver)
+    {
+        $this->controllerResolver = $controllerResolver;
+    }
+
+    public function onKernelRequest(AttributeRequestEvent $event)
+    {
+        $request = $event->getRequest();
+
+        if (
+            method_exists(\ReflectionProperty::class, 'getAttributes') &&
+            $controller = $this->controllerResolver->getController($request)
+        ) {
+            $class = new \ReflectionClass($controller[0]);
+            $method = $class->getMethod($controller[1]);
```

Comment:
according to the signature, `getController` returns any callable. some application might do different than class methods... to be on the safe side, i would rewrite the start of this method as:

```
$request = $event->getRequest();
$controller = $this->controllerResolver->getController($request);

if (!is_array($controller) || 2 !== count($controller)) {
    return;
}

$class = ...
```